### PR TITLE
XML Schema

### LINKED ISSUE #36: XML Schema for LOGOS
**Is your feature request related to a problem? Please describe.**
Add XML schema for LOGOS to validate XML input

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
close #36 

##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual.
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Regression tests have to complete successfully.
- [x] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test.
- [x] 6. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.

### CODE REVIEW

**Comment by wangcj05** [comment]
@mandd This PR is ready for you to review.

File: XSDSchemas/validate_xml.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
```
Comment (mandd) [inline XSDSchemas/validate_xml.py]: I think this text refers to RAVEN :-P

File: XSDSchemas/get_coverage_tests.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
```
Comment (mandd) [inline XSDSchemas/get_coverage_tests.py]: Check text

File: XSDSchemas/get_coverage_tests.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+
+logosDir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+testDir = os.path.join(logosDir, 'tests', 'CapitalInvestments')
+# testDir = os.path.join(logosDir, 'tests')
```
Comment (mandd) [inline XSDSchemas/get_coverage_tests.py]: remove comment line?

File: XSDSchemas/get_coverage_tests.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    for spec in testFileList:
+      # check if test is skipped or an executable is required
+      if "required_executable" in spec or "skip" in spec:
+        continue
+      if "input" not in spec:
+        continue
+      testType = spec.get('type', "notfound").strip()
+      newTest = spec['input'].split()[0]
+
+      if whichTests in [1]:
```
Comment (mandd) [inline XSDSchemas/get_coverage_tests.py:67]: is this statement correct?

File: XSDSchemas/Settings.xsd [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    <xsd:complexType name="SettingsData">
+        <xsd:all>
+            <xsd:element name="mandatory" type="xsd:string"   minOccurs="0"/>
+            <xsd:element name="nonSelection" type="LogosBool"  minOccurs="0"/>
+            <xsd:element name="consistentConstraintI" type="LogosBool"     minOccurs="0"/>
+            <xsd:element name="consistentConstraintII" type="LogosBool"   minOccurs="0"/>
+            <xsd:element name="solver" type="xsd:string"   minOccurs="0"/>
+            <xsd:element name="workingDir" type="xsd:string"   minOccurs="0"/>
+            <xsd:element name="tee" type="LogosBool"   minOccurs="0"/>
```
Comment (mandd) [inline XSDSchemas/Settings.xsd:12]: is "tee" correct?

File: XSDSchemas/validate_xml.py (hunk 1 above)
Comment (wangcj05) [reply XSDSchemas/validate_xml.py]: fixed, I copied it from RAVEN one, and modified it for LOGOS

File: XSDSchemas/get_coverage_tests.py (hunk 2 above)
Comment (wangcj05) [reply XSDSchemas/get_coverage_tests.py]: fixed

File: XSDSchemas/get_coverage_tests.py (hunk 3 above)
Comment (wangcj05) [reply XSDSchemas/get_coverage_tests.py]: removed

File: XSDSchemas/get_coverage_tests.py (hunk 4 above)
Comment (wangcj05) [reply XSDSchemas/get_coverage_tests.py:67]: This is ok. 'whichTests' can either be 1 or 2.

File: XSDSchemas/Settings.xsd (hunk 5 above)
Comment (wangcj05) [reply XSDSchemas/Settings.xsd:12]: Yes, it is used by the solver to output the solver information.

**Comment by wangcj05** [comment]
@mandd I have resolved your comments, please let me know if you have additional comments.

**Review by mandd — APPROVED** [review]

**Pull request merged**